granular_flow.go

```diff
@@ -222,47 +222,47 @@ func (c *Client) execute(
 	tp reflect.Type,
 	cdcs codecPair,
 ) error {
-	buf := buff.New(c.buffer[:0])
-	buf.BeginMessage(message.Execute)
-	buf.PushUint16(0)       // no headers
-	buf.PushBytes([]byte{}) // no statement name
-	cdcs.in.Encode(buf, q.args)
-	buf.EndMessage()
+	c.buf.Reset()
```

Comment:
Hm.... is this safe? What if there are some unprocessed (asynchronously received) messages in the buf? We'll have to deal with those soon. 

I don't think using a new buffer each time you want to compose a message is a noticeable performance hit, but it's certainly safer and easier to reason about.

Reply:
This change is not meant to be the long term strategy. It is tidying up some inconsistencies. Previously prepare allocated on every call, while optimistic reused reallocated memory. 

For now this is safe because buffers are not used concurrently, even when they are used concurrently it is easy to ensure that only one go routine has access to a given buffer using channels.

Reply:
If this makes refactoring is temporary and makes your life easier - then sure. But we'll need to revise this.

Reply:
Although I still feel very uneasy about this. It would make more sense to add two separate buffers to Client: `readBuf` and `writeBuf`